src/main/kotlin/uk/gov/justice/digital/hmpps/bodyscan/apiclient/BodyScanPrisonApiClient.kt

```diff
@@ -1,16 +1,29 @@
 package uk.gov.justice.digital.hmpps.bodyscan.apiclient
 
 import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.ClientResponse
 import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.bodyscan.apiclient.model.PersonalCareCounter
+import uk.gov.justice.digital.hmpps.bodyscan.apiclient.model.PersonalCareNeeds
+import uk.gov.justice.digital.hmpps.bodyscan.apiclient.model.SentenceDetails
+import uk.gov.justice.digital.hmpps.config.ClientErrorResponse
+import uk.gov.justice.digital.hmpps.config.ClientException
+import uk.gov.justice.digital.hmpps.config.NotFoundException
 import uk.gov.justice.digital.hmpps.config.typeReference
 import java.time.LocalDate
 import java.time.format.DateTimeFormatter
 
 @Component
 class BodyScanPrisonApiClient(@Qualifier("prisonApiWebClient") private val webClient: WebClient) {
 
+  fun propagateClientError(response: ClientResponse, message: String): Mono<ClientException> =
```

Comment:
maybe it's better to not propagate the exception? there aren't currently any expected scenarios where we would want to show the user a custom message.. I think the stand error page should suffice?

Reply:
sorted